### PR TITLE
enhance logging 

### LINKED ISSUE #49: Improve agent logging facitilites
Currently the agent only logs to STDERR/STDOUT and it is hard to parse and determine the severity of the log entries. As user I would like to be able to configure the log level and format of the agents run time logs (JSON).

### PR DESCRIPTION
**Pull request by DavidXArnold**
#### What does this PR do?
- adds the ability to specify log_format & log_level via configuration flags
- updates dependencies
- refactors util.CheckRequiredSettings
- adds a version command line flag

#### Where should the reviewer start?
[here](https://github.com/cloudability/metrics-agent/compare/master...DavidXArnold:49?expand=1#diff-ff7686b39bf90dc2520886fb874371a4R18)

#### Any background context you want to provide?
This PR enhances the agents logging facilities so that message severity is now specified in every log entry as well as the ability to specify a desired logging level.  Additionally the ability to specify the log format as either JSON or plain text has been added.

#### What are the relevant Github Issues?
Fixes #49 

#### Developer Done List

- [x] Tests Added/Updated
- [x] Updated README.md
- [x] Verified backward compatible
- [x] Verified database migrations will not be catastrophic
- [x] Considered Security, Availability and Confidentiality

#### Reviewer/Approval Done List

- [x] Tests Pass Locally
- [x] CI Build Passes
- [x] Verified README.md is updated
- [x] Verified changes are backward compatible
- [x] Reviewed impact to Security, Availability and Confidentiality (if issue found, add comments and request changes)

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -11,6 +11,10 @@
...
-[[override]]
+[[constraint]]
   name = "github.com/spf13/cobra"
-  revision = "d5bde60e022436203e377c47806e001418c16af3"
\ No newline at end of file
+  version = "v0.0.5"
+
+[[override]]
+  name = "github.com/sirupsen/logrus"
+  revision = "07a84ee7412e7a28663d92930a1d46f81b124ee1"
```
Comment (mnorbury) [inline Gopkg.toml]: Could this be a version rather than a commit e.g. 1.4.2?

Comment (DavidXArnold) [reply Gopkg.toml]: Great question, unfortunately they have not yet cut a version that includes the PadLevelText option to the TextFormatter . This has been merged to the logrus master branch just not out in a release yet. Once the next release is out we can switch to that.

Comment (mnorbury) [reply Gopkg.toml]: Ah, okay. In that case using the revision is fine.
